app/controllers/discovered_hosts_controller.rb

```diff
@@ -38,14 +38,11 @@ def show
     @range = nil
     # summary report text
     @report_summary = nil
-    init_regex_and_categories
+    @categories_names = [N_("Highlights"), N_("Storage"), N_("Hardware"), N_("Network"), N_("Software"), N_("IPMI"), N_("Miscellaneous")]
     @interfaces = []
     get_interfaces
-    @host.facts_hash.each do |key, value|
-      value = number_to_human_size(value) if /size$/.match(key)
-      assign_fact_to_category(key, value)
-    end
-    add_custom_facts
+    resolver = ForemanDiscovery::FactToCategoryResolver.new(@host)
```

Comment:
I took the approach where the new class will only need a `@host` object and it will take care of rest. I think that this way will improve readability and things around facts_to_category will be on place - instead of last implementation.

Reply:
Absolutely, object composition is my preferred approach in most object-oriented languages including Ruby. Thanks for not creating a concern for this.